### DISCUSSION TITLE
Zigbee2MQTT - Error while starting zigbee-herdsman

### DISCUSSION THREAD

**Discussion by mefrasa** [indomus/forum#2267] [HUB: Home Assistant]
Ciao a tutti,
sto provando in tutti i modi ad avviare Zigbee2MQTT  ma non vuole saperne.
Il log è il seguente: 

![immagine](https://user-images.githubusercontent.com/44811375/211581630-b46d772a-ccb6-46cb-90c4-0811c72eb6cb.png)

La configurazione, questa: 
![immagine](https://user-images.githubusercontent.com/44811375/211581809-fd1100fd-d8a3-49e4-98a2-7a7f052b2fd2.png)

Mosquito pare funzionare...
![immagine](https://user-images.githubusercontent.com/44811375/211582009-c3f151e2-6c2a-4854-b001-69988bb04063.png)

Chi può darmi una mano?
Grazie

**Comment by Newreader** [comment]
Prova a sostituire nella configurazione mqtt la voce
server:  //mqtt/127.0.0.1 
con
server: //mqtt/Ip del broker mqtt:porta

**Reply by mefrasa** [reply]
Grazie,
ho appena provato, sia con la 1883 che con la 8883, ma nulla :(

**Reply by Newreader** [reply]
Hai sostituito anche l'indirizzo IP 127.0.0.1?

**Reply by mefrasa** [reply]
Ma essendo mosquito broker installato come addon l'indirizzo IP dovrebbe essere corretto

**Reply by Newreader** [reply]
Tu hai installato ZigBee2MQTT su Home Assistant OS/Supervised?

**Reply by mefrasa** [reply]
Si, ho seguito la guida presente su inDomus

**Comment by Newreader** [comment]
Io non ho la tua stessa installazione, tuttavia ti consiglio di rileggere come "Configurare ZigBee2MQTT" nella guida:
https://indomus.it/guide/come-installare-e-configurare-zigbee2mqtt-su-home-assistant-hassio/
 e i relativi parametri di configurazione (es. server)

**Reply by mefrasa** [reply]
E' quella che ho seguito... ma purtroppo non ne vengo a capo... e neanche i log aiutano

**Comment by Newreader** [comment]
Quello a cui facevo riferimento nella guida è riportato sotto:
"I campi da personalizzare sono, principalmente: “server“, “user” e “password” e infine “port“.

Nel primo (“server“) va indicato il nostro broker MQTT: se si usa un proprio broker presente sulla rete, è ovviamente necessario indicare in questo campo l’indirizzo IP corretto; diversamente, se si è seguita la nostra guida per installare Mosquitto Broker come add-on Home Assistant OS/Supervised, allora è necessario lasciare semplicemente “mqtt://core-mosquitto“.

Nei campi “user” e “password“, invece, è necessario indicare le credenziali d’accesso al broker MQTT, indipendentemente da dove esso sia in esecuzione. Infine, nel campo “port” indicare la porta logica dell’antenna [identificata in precedenza](https://indomus.it/guide/come-installare-e-configurare-zigbee2mqtt-su-home-assistant-hassio/#port)."
Inoltre l'indirizzo IP a cui facevo riferimento è quello della macchina che ospita MQTT (ad esempio quello del Raspberry)

**Reply by mefrasa** [reply]
Le ho provate tutte, secondo la guida dovrebbe essere così... 

![immagine](https://user-images.githubusercontent.com/44811375/211625686-ff2dc9f9-1e6a-43d3-bee3-4080c17fb086.png)

**Reply by Newreader** [reply]
Come già ti scrivevo prova a sostituire //core-mosquitto con l'indirizzo del raspberry (esempio //192.168.1.50 o simile)

**Reply by mefrasa** [reply]
Non ne vuole sapere

**Comment by popeye70** [comment]
Ciao a tutti,
ho installato il Zigbee2MQTT nel mio QNAP seguendo le guide di Indomus, dopo alcune settimane di funzionamento regolare improvvisamente (probabilmente dopo un riavvio del NAS) l'ho trovato spento e ora non vuole più saperne di riavviarsi:

[16:57:01] INFO: Preparing to start...
[16:57:02] INFO: Socat not enabled
[16:57:03] INFO: Starting Zigbee2MQTT...
Zigbee2MQTT:info  2023-04-08 16:57:11: Logging to console and directory: '/config/zigbee2mqtt/log/2023-04-08.16-57-10' filename: log.txt
Zigbee2MQTT:info  2023-04-08 16:57:11: Starting Zigbee2MQTT version 1.30.3 (commit #unknown)
Zigbee2MQTT:info  2023-04-08 16:57:11: Starting zigbee-herdsman (0.14.103)
Zigbee2MQTT:error 2023-04-08 16:57:11: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2023-04-08 16:57:11: Failed to start zigbee
Zigbee2MQTT:error 2023-04-08 16:57:11: Check https://www.zigbee2mqtt.io/guide/installation/20_zigbee2mqtt-fails-to-start.html for possible solutions
Zigbee2MQTT:error 2023-04-08 16:57:11: Exiting...
Zigbee2MQTT:error 2023-04-08 16:57:12: Error: Error while opening serialport 'Error: Error: No such file or directory, cannot open /dev/serial/by-id/usb-Silicon_Labs_Sonoff_Zigbee_3.0_USB_Dongle_Plus_0001-if00-port0'
    at SerialPort.<anonymous> (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/znp/znp.ts:146:28)
    at SerialPort._error (/app/node_modules/@serialport/stream/dist/index.js:76:22)
    at /app/node_modules/@serialport/stream/dist/index.js:112:18

Questa invece è la sua configurazione:

data_path: /config/zigbee2mqtt
socat:
  enabled: false
  master: pty,raw,echo=0,link=/tmp/ttyZ2M,mode=777
  slave: tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5
  options: "-d -d"
  log: false
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://core-mosquitto
  user: mqtt_user
  password: 
  version: 5
  include_device_information: true
serial:
  port: >-
    /dev/serial/by-id/usb-Silicon_Labs_Sonoff_Zigbee_3.0_USB_Dongle_Plus_0001-if00-port0

Qualcuno gentilmente riesce a suggerirmi cosa/dove controllare?

**Reply by snakuzzo** [reply]
```
Error: Error while opening serialport 'Error: Error: No such file or directory, cannot open /dev/serial/by-id/usb-Silicon_Labs_Sonoff_Zigbee_3.0_USB_Dongle_Plus_0001-if00-port0'
```

La chiavetta non viene vista, o comunque non viene vista su quella porta...probabilmente la porta è cambiata dopo il reboot

**Comment by popeye70** [comment]
Ciao,
e innanzitutto grazie di avermi risposto.
Il mio HA gira su Vitualization Station, mentre Mosquitto e Zigbee2MQTT sono montate in Container Station via Docker.
Sì è molto probabile che sia stato il reboot del QNAP a causare il problema, tanto che ora un nuovo riavvio totale del sistema lo ha magicamente risolto.
Il fatto è che al riavvio il NAS si riappropria delle USB, compresa quella con inserito il dongle Zigbee...e quindi una volta ripartito il QNAP occorre innanzitutto riassegnare la USB alla VM di HA e poi avviare manualmente anche HA stesso.
Che tu sappia, c'è modo al reboot del QNAP di far ripartire automaticamente anche HA con la sua USB, mantenendo pure la porta precedentemente assegnata?
Grazie ancora, e con l'occasione, buona Pasqua a tutti.
